subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/ArtifactTransformRegistrationsInternal.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts.transform;
+
+import org.gradle.api.Transformer;
+import org.gradle.api.artifacts.ResolvedArtifact;
+import org.gradle.api.artifacts.transform.ArtifactTransformRegistrations;
+import org.gradle.api.attributes.AttributeContainer;
+
+import java.io.File;
+import java.util.List;
+
+public interface ArtifactTransformRegistrationsInternal extends ArtifactTransformRegistrations {
```

Comment:
I don't understand this interface. I think the cache should be part of the transformer (which is also Project scoped) instead of having this broad caching interface.